### ISSUE TITLE
Include tests

### ISSUE THREAD

**Issue by alestiago** [jonasroussel/dart_jsonwebtoken#47]
## Description

There are currently no tests, as a user of the package I would like the package to be well and fully tested. Mainly to ensure the code consistently works and no unexpected breaking changes are pushed.

## Requirements

- [ ] Add tests to the codebase
- [ ] Reach 100% test coverage
- [ ] Include workflow to automatically check all tests are passing

## Further context

This package is currently being recommend as [part of a Dart Frog tutorial](https://dartfrog.vgv.dev/docs/tutorials/jwt_authentication#protecting-the-authentication-token). It would be ideal if this package had test to guarantee a certain level of quality.

---

Thanks for the package, would you be willing to accept external contributions that add tests to the codebase?

**Comment by jonasroussel** [comment]
Hey @alestiago !

I didn't know that Dart Frog recommended this package, very glad to hear it!

External contributions are always welcome! I'll try to set up a some tests by the end of the week and after that you'll be able to add your own tests to reach 100% of test coverage.

I've already started a workflow to auto-deploy the package, at the same time I'll make one more to run the tests.

**Comment by jonasroussel** [comment]
@alestiago

I wanted to let you know that I've just set up the first tests. These are still basic tests that only test the 2 main functions `sign` & `verify` with all available algorithms but I may add more soon, you're also free to add your own.

In the meantime, I've created 2 GH actions:
one that will analyze the code (formatting & litering) and run tests for each new PR and PR change,
and another that will publish the package on the new tags (also running the CI analyze and test beforehand).
